packages/core/src/dialects/cockroachdb/query-generator.ts

```diff
@@ -0,0 +1,116 @@
+      'JOIN information_schema.KEY_COLUMN_USAGE  cu',
+      'ON tc.table_schema=cu.table_schema and tc.table_name=cu.table_name',
+      'and tc.constraint_name=cu.constraint_name',
+      `and tc.constraint_type='PRIMARY KEY') pk`,
+      'ON pk.table_schema=c.table_schema',
+      'AND pk.table_name=c.table_name',
+      'AND pk.column_name=c.column_name',
+      `WHERE c.table_name = ${this.escape(table.tableName)}`,
+      `AND c.table_schema = ${this.escape(table.schema)}`,
+    ]);
+  }
+
+  truncateTableQuery(tableName: string, options?: TruncateTableQueryOptions): string {
+    if (options) {
+      rejectInvalidOptions(
+        'truncateTableQuery',
+        this.dialect.name,
+        TRUNCATE_TABLE_QUERY_SUPPORTABLE_OPTIONS,
+        TRUNCATE_TABLE_QUERY_SUPPORTED_OPTIONS,
+        options,
+      );
+    }
+
+    return [
+      `TRUNCATE ${this.quoteTable(tableName)}`,
+      options?.cascade ? ' CASCADE' : '',
+    ].join('');
+  }
+
+  // Overriding postgres implementation since Postgres uses DO BEGIN which is not supported by cockroachdb
+  pgEnum<Members extends string>(tableName: string, attr: string, dataType: ENUM<Members>, options: any) {
+    const enumName = this.pgEnumName(tableName, attr, options);
+    let values;
+
+    if (dataType instanceof ENUM && dataType.options.values) {
+      values = `ENUM(${dataType.options.values.map(value => this.escape(value)).join(', ')})`;
+    } else {
+      values = dataType.toString().match(/^ENUM\(.+\)/)?.[0];
+    }
+
+    let sql: string = `CREATE TYPE IF NOT EXISTS ${enumName} AS ${values};`;
+    if (Boolean(options) && options.force === true) {
+      sql = this.pgEnumDrop(tableName, attr) + sql;
+    }
+
+    return sql;
+  }
+
+  dropSchemaQuery(schema: string, options?: DropSchemaQueryOptions): string {
+    if (schema === 'crdb_internal') {
+      throw new Error('Cannot remove crdb_internal schema in Cockroachdb');
+    }
+
+    return super.dropSchemaQuery(schema, options);
+  }
+
+  showIndexesQuery(tableName: TableNameOrModel): string {
+    const table = this.extractTableDetails(tableName);
+
+    // TODO [>=6]: refactor the query to use pg_indexes
+    return joinSQLFragments([
+      'SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, MAX(ix.indnkeyatts) as index_fields_length,',
+      'array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names,',
+      'pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a , pg_namespace s',
+      'WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND',
+      `t.relkind = 'r' and t.relname = ${this.escape(table.tableName)}`,
+      `AND s.oid = t.relnamespace AND s.nspname = ${this.escape(table.schema)}`,
+      'GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey, ix.indnkeyatts ORDER BY i.relname;',
+    ]);
+  }
+
+  getUuidV1FunctionCall(): string {
```

Comment:
`getUuidV4FunctionCall` is fine as it has a different implementation, but `getUuidV1FunctionCall` is the same implementation so this method can be removed